File: backend/app/core/response.py
from typing import Any


def success(data: Any = None, message: str = "OK"):
    return {
        "success": True,
        "code": 0,
        "message": message,
        "data": data,
    }


def error(message: str = "Error", code: int = -1):
    return {
        "success": False,
        "code": code,
        "message": message,
        "data": None,
    }

File: packages/utils/src/response.ts
import type { ApiResponse } from '@howard-aios/types';

export function success<T>(data: T, message = 'OK'): ApiResponse<T> {
  return {
    success: true,
    code: 0,
    message,
    data,
  };
}

export function error(message = 'Error', code = -1): ApiResponse<null> {
  return {
    success: false,
    code,
    message,
    data: null,
  };
}
